staging/src/k8s.io/kubectl/pkg/describe/describe.go

```diff
@@ -3523,7 +3523,10 @@ func (d *NodeDescriber) Describe(namespace, name string, describerSettings Descr
 			klog.Errorf("Unable to construct reference to '%#v': %v", node, err)
 		} else {
 			// TODO: We haven't decided the namespace for Node object yet.
-			ref.UID = types.UID(ref.Name)
+			// there are two UIDs for host events:
+			// controller use node.uid
+			// kubelet use node.name
+			ref.UID = ""
```

Comment:
If the discrepancy is in the kubelet which is injecting name into UID that should be fixed on the kubelet side of things. What I'm willing to accept on the kubectl side of things are 2 `searchEvents` calls one using UID and other using name such that we'll scrape both events and leave a TODO here pointing to issue against kubelet, such that we'll be able to remove the workaround after kubelet is fixed.

Reply:
@soltysh 
I have a question, by what field are the events sorted, as the merge needs to combine them and reorder them, is it the creation time field?